pygmt/src/grdgradient.py

```diff
@@ -0,0 +1,118 @@
+"""
+grdgradient - Create a gradient from a grid.
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    args_in_kwargs,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    A="azimuth",
+    D="direction",
+    E="radiance",
+    G="outgrid",
+    R="region",
+    V="verbose",
+    n="interpolation",
+)
+@kwargs_to_strings(A="sequence", E="sequence", R="sequence")
+def grdgradient(grid, **kwargs):
+    r"""
+    Compute the directional derivative of the vector gradient of the data.
+
+    Can accept ``azimuth``, ``direction``, and ``radiance`` input to create
+    the resulting gradient.
+
+    Full option list at :gmt-docs:`grdgradient.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    azimuth : int or float or str or list or xarray.DataArray
```

Comment:
I don't think `azimuth` supports xarray.DataArray.
```suggestion
    azimuth : int or float or str or list
```